src/test/regress/expected/citus_local_tables_queries.out

```diff
@@ -0,0 +1,894 @@
+NOTICE:  executing the copy locally for shard xxxxx
+INSERT INTO reference_table
+SELECT reference_table.* from reference_table JOIN citus_local_table ON (true) LIMIT 10;
+NOTICE:  executing the command locally: SELECT reference_table.a, reference_table.b FROM (citus_local_table_queries.reference_table_1509002 reference_table JOIN citus_local_table_queries.citus_local_table_1509000 citus_local_table ON (true)) LIMIT 10
+NOTICE:  executing the copy locally for shard xxxxx
+INSERT INTO citus_local_table
+SELECT reference_table.* from reference_table JOIN distributed_table ON (true) LIMIT 10;
+NOTICE:  executing the copy locally for shard xxxxx
+INSERT INTO distributed_table
+SELECT reference_table.* from reference_table JOIN citus_local_table ON (true) LIMIT 10;
+NOTICE:  executing the command locally: SELECT reference_table.a, reference_table.b FROM (citus_local_table_queries.reference_table_1509002 reference_table JOIN citus_local_table_queries.citus_local_table_1509000 citus_local_table ON (true)) LIMIT 10
+INSERT INTO postgres_local_table
+SELECT reference_table.* from reference_table JOIN citus_local_table ON (true) LIMIT 10;
+NOTICE:  executing the command locally: SELECT reference_table.a, reference_table.b FROM (citus_local_table_queries.reference_table_1509002 reference_table JOIN citus_local_table_queries.citus_local_table_1509000 citus_local_table ON (true)) LIMIT 10
+-- not supported joins
+INSERT INTO citus_local_table
+SELECT distributed_table.* from distributed_table JOIN citus_local_table ON (true) LIMIT 10;
+ERROR:  citus local tables cannot be used in distributed queries
+-- when wrapped into a CTE, join works fine
+INSERT INTO citus_local_table
+SELECT distributed_table.* from distributed_table JOIN (WITH cte AS (SELECT * FROM citus_local_table) SELECT * FROM cte) as foo ON (true) LIMIT 10;
+NOTICE:  executing the command locally: SELECT a, b FROM citus_local_table_queries.citus_local_table_1509000 citus_local_table
+NOTICE:  executing the copy locally for shard xxxxx
+-- multi row inserts
+INSERT INTO citus_local_table VALUES (1, 2);
+NOTICE:  executing the command locally: INSERT INTO citus_local_table_queries.citus_local_table_1509000 (a, b) VALUES (1, 2)
+INSERT INTO postgres_local_table
+SELECT * from citus_local_table LIMIT 10;
+NOTICE:  executing the command locally: SELECT a, b FROM citus_local_table_queries.citus_local_table_1509000 citus_local_table LIMIT 10
+-- INSERT SELECT into distributed table from
+-- citus local table is okay
+INSERT INTO distributed_table
+SELECT * from citus_local_table LIMIT 10;
+NOTICE:  executing the command locally: SELECT a, b FROM citus_local_table_queries.citus_local_table_1509000 citus_local_table LIMIT 10
+---------------------------------------------------------------------
+----- DELETE / UPDATE -----
+---------------------------------------------------------------------
+-- no direct joins supported
+DELETE FROM citus_local_table
```

Comment:
This feels interesting. The joins between citus local tables and postgres local tables work expected. Why not modifications?

Reply:
(talked in person and decided not to allow such modifications for now)